specs/unmanage_cluster.adoc

```diff
@@ -0,0 +1,373 @@
+
+* If a previous import failed or cluster is in mis-configured state after import
+(import failed with errors field not populated for cluster), the import and
+un-manage both the options would be enabled in UI. If user selects the import
+option now, it lands in import cluster view/page. If there was a previous import
+failed, then modal dialog shows up and message would be something like `Import
+cluster previously failed with <job_id>. Before import, you need to correct the
+issues and then un-manage the cluster`. This dialog has `Ok` and `Cancel`
+buttons.
+
+* If un-manage fails, it would provide a tooltip/info with failure message `If
+un-manage fails, resolve the issue and then try un-manage cluster again`. It
+would show a message to say `Unmanage Cluster` failed having a `View Details`
+hyperlink in the cluster list view.
+
+
+=== Security impact:
+
+None
+
+=== Other end user impact:
+
+User gets an option to un-manage an existing cluster and can re-import at later
+stage
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+The tendrl-ansible module need to provide a mechanism to setup tendrl components
+and dependencies on additional new node in the cluster.
+
+<TBD> details to be added here of the plyabooks etc.
+
+=== Developer impact:
+
+None
+
+
+== Implementation:
+
+* https://github.com/Tendrl/commons/issues/797
+
+
+=== Assignee(s):
+
+Primary assignee:
+  shtripat
+  mbukatov
+  a2batic
+
+=== Work Items:
+
+* https://github.com/Tendrl/specifications/issues/252
+
+
+== Dependencies:
+
+* https://github.com/Tendrl/api/issues/349
+
+== Testing:
+
+* Check if UI dashboard has an option to trigger un-manage cluster flow
+
+* Check if the flow gets completed successfully and verify if the grafana
+dashboard reflects and cluster details available now for the selected cluster
+
+* Verify that no grafana alert dashboards available now for the un-managed
+cluster
+
+* Verify that the clusters list report the cluster as un-managed and import
+option is enabled now
+
+* Try to import the cluster back and it should be successful. All grafana
+dashboards, grafana alert dashboards and UI reflect the cluster details back
+
+* Invoke the REST end point `clusters/{int-id}/unmanage` and the cluster should
+be un-managed successfully
+
+* On un-manage cluster completion, the alert dashboards in grafana would vanish
+for the entities of the cluster like volume, bricks etc. Verify to make sure the
+same happens as expected
+
+* Once cluster is un-managed the details of the cluster would vanish from
+dashboards in grafana. Verify the same happens as expected
+
+* Verify that the final alert post un-manage flow, tells about removal of
+details from grafana dashboards and grafana alert dashboards
+
+* Verify the scenatio when a cluster import fails, and user is able to start
+a un-manage + reimport cluster option from UI. UI should be able to list details
+of both the tasks in this scenario
+
```

Comment:
There's one other scenario which should be tested and that's when un-manage fails.
Moreover check of View Details link should be here too - it should be present during un-manage task run and if it fails. It should provide almost the same as for import.